specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json

```diff
@@ -532,6 +578,115 @@
           }
         }
       }
+    },
+    "ManagedRuleSet": {
+      "type": "object",
+      "description": "Defines a managed rule set.",
+      "required": [
+        "ruleSetType",
+        "ruleSetVersion"
+      ],
+      "properties": {
+        "ruleSetType": {
+          "description": "Defines the rule set type to use.",
+          "type": "string"
+        },
+        "ruleSetVersion": {
+          "description": "Defines the version of the rule set to use.",
+          "type": "string"
+        },
+        "ruleGroupOverrides": {
+          "description": "Defines the rule group overrides to apply to the rule set.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleGroupOverride"
+          }
+        }
+      }
+    },
+    "ManagedRuleGroupOverride": {
+      "description": "Defines a managed rule group override setting.",
+      "required": [
+        "ruleGroupName"
+      ],
+      "properties": {
+        "ruleGroupName": {
+          "description": "Describes the managed rule group to override.",
+          "type": "string"
+        },
+        "rules": {
+          "description": "List of rules that will be disabled. If none specified, all rules in the group will be disabled.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleOverride"
+          }
+        }
+      }
+    },
+    "ManagedRuleOverride": {
+      "description": "Defines a managed rule group override setting.",
+      "required": [
+        "ruleId"
+      ],
+      "properties": {
+        "ruleId": {
+          "description": "Identifier for the managed rule.",
+          "type": "string"
+        },
+        "state": {
+          "description": "Describes if the managed rule is in enabled or disabled state. Defaults to Disabled if not specified.",
+          "type": "string",
+          "enum": [
+            "Disabled"
```

Comment:
Shouldn't there also be `Enabled`?

Reply:
No. By default every rule is enabled. when user wants to disable an rule, a Disabled entry has to be added. The "Disabled" enum is introduced so as to keep the config in parity with AFD Waf.

Reply:
In `WafPolicyCreateOrUpdate.json` state is set as `Enabled` - if the customer can pass that value and it's going to be accepted, it should be added to Swagger.

Reply:
Also, there's a cotradiction. In property description it says `Defaults to Disabled if not specified.` but you say `By default every rule is enabled`

Reply:
Talked in Teams, resolved, description will be updated to avoid confusion

Reply:
@number213, As discussed  i will update the description. 

Just to summarize: for us all the rules  are enabled. If an entry is there to be overriden it means that user wants to disable it. Thats why we dont have a enabled entry. The reason we have the disabled is for the config to look similar to AFD Waf.